cloud-sql/r2dbc/README.md

```diff
@@ -0,0 +1,76 @@
+# Connecting to Cloud SQL - MySQL and Postgres
+
+## Before you begin
+
+1. If you haven't already, set up a Java Development Environment (including google-cloud-sdk and
+maven utilities) by following the [Java setup guide](https://cloud.google.com/java/docs/setup) and
+[creating a project](https://cloud.google.com/resource-manager/docs/creating-managing-projects#creating_a_project).
+
+1. You can use MySQL or PostgreSQL instance for this sample. 
+Create a 2nd Gen Cloud SQL Instance by following corresponding instructions: 
+[MySQL](https://cloud.google.com/sql/docs/mysql/create-instance) /
+[PostgreSQL](https://cloud.google.com/sql/docs/postgres/create-instance).
+Note the connection string, database user, and database password that you create. 
+
+1. Create a database for your application by following corresponding instructions:
+[MySQL](https://cloud.google.com/sql/docs/mysql/create-manage-databases) /
+[PostgreSQL](https://cloud.google.com/sql/docs/postgres/create-manage-databases).
+Note the database name.
+
+1. Edit `src/main/resources/application.properties` and enter your connection details in the following format
+    ```
+    connectionString = r2dbc:gcp:<'mysql' or 'postgres'>://<user>:<password>@<connection_name>/<db_name>
+    ```
+    Example for MySQL:
+    ```
+    connectionString = r2dbc:gcp:mysql://user:123456@my-project:us-central1:r2dbctest/testdb 
+    ``` 
+
+    Example for PostgreSQL:
+    ```
+    connectionString = r2dbc:gcp:postgres://user:123456@my-project:us-central1:r2dbctest/testdb 
+    ``` 
+## Schema
+
+The schema will be created automatically when the application starts.
+
+## Deploying locally
+
+To run this application locally, run the following command inside the project folder:
+
+```sh
+mvn spring-boot:run
+```
+
+Navigate to `http://localhost:8080` to verify your application is running correctly.
+
```

Comment:
Could we also add instructions for deploying to App Engine? (Or is there a reason why we don't have them?)

Reply:
done